bodhi/tests/client/test_bindings.py

```diff
@@ -987,3 +987,71 @@ def test___unicode__(self):
         self.assertEqual(unicode(exc.update), u'bodhi-2.2.4-1.el7')
         self.assertEqual(type(unicode(exc.update)), unicode)
         self.assertEqual(unicode(exc), 'Update not found: bodhi-2.2.4-1.el7')
+
+
+class TestBodhiClient_parse_file(unittest.TestCase):
+    """
+    Test the BodhiClient.parse_file() method.
+    """
+    def test_parsing_valid_file(self):
+        """
+        Test parsing a valid update template file
+        """
+        s = ("[fedora-workstation-backgrounds-1.1-1.fc26]\n"
+             "# bugfix, security, enhancement, newpackage (required)\n"
+             "type=bugfix\n"
+             "\n"
+             "# testing, stable\n"
+             "request=testing\n"
+             "\n"
+             "# Bug numbers: 1234,9876\n"
+             "bugs=123456,43212\n"
+             "\n"
+             "# Here is where you give an explanation of your update.\n"
+             "notes=Initial Release\n"
+             "\n"
+             "# Enable request automation based on the stable/unstable karma thresholds\n"
+             "autokarma=True\n"
+             "stable_karma=3\n"
+             "unstable_karma=-3\n"
+             "\n"
+             "# Automatically close bugs when this marked as stable\n"
+             "close_bugs=True\n"
+             "\n"
+             "# Suggest that users restart after update\n"
+             "suggest_reboot=False")
+
+        update_template = open("/tmp/bodhi-test-parsefile1", "w")
+        update_template.write(s)
+        update_template.close()
```

Comment:
You could do this without writing a real file to the disk by mocking the builtin ```open()``` function. Here's an example where I did that to simulate opening an empty file:

https://github.com/fedora-infra/bodhi/blob/2.7.0/bodhi/tests/client/test_bindings.py#L98-L118

Some of the other tests in that class substitute other data in the file. This is a nice way to do it because you don't have to worry about cleaning up the file if the test fails, and it performs a little better too!